contrib/world_builder/include/world_builder/utilities.h

```diff
@@ -0,0 +1,181 @@
+  This file is part of the World Builder.
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU Lesser General Public License as published
+   by the Free Software Foundation, either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef _world_builder_utilities_h
+#define _world_builder_utilities_h
+
+#include <vector>
+
+#include <world_builder/point.h>
+#include <world_builder/coordinate_system.h>
+#include <world_builder/coordinate_systems/interface.h>
+
+
+namespace WorldBuilder
+{
+
+  namespace CoordinateSystems
+  {
+    class Interface;
+  }
+  namespace Utilities
+  {
+
+    /**
+     * Given a 2d point and a list of points which form a polygon, computes if the point
+     * falls within the polygon.
+     */
+    bool
+    polygon_contains_point(const std::vector<Point<2> > &point_list,
+                           const Point<2> &point);
+
+    /**
+     * Given a 2d point and a list of points which form a polygon, compute the smallest
+     * distance of the point to the polygon. The sign is negative for points outside of
+     * the polygon and positive for points inside the polygon.
+     */
+    double
+    signed_distance_to_polygon(const std::vector<Point<2> > &point_list_,
+                               const Point<2> &point_);
+
+
+    /*
+    * A class that represents a point in a chosen coordinate system.
+    */
+    class NaturalCoordinate
+    {
+      public:
+        /**
+         * Constructor based on providing the geometry model as a pointer
+         */
+        NaturalCoordinate(const std::array<double,3> &position,
+                          const ::WorldBuilder::CoordinateSystems::Interface &coordinate_system);
+
+        /**
+         * Constructor based on providing the geometry model as a pointer
+         */
+        NaturalCoordinate(const Point<3> &position,
+                          const ::WorldBuilder::CoordinateSystems::Interface &coordinate_system);
+
+        /**
+         * Returns the coordinates in the given coordinate system, which may
+         * not be Cartesian.
+         */
+        std::array<double,3> &get_coordinates();
+
+        /**
+         * The coordinate that represents the 'surface' directions in the
+         * chosen coordinate system.
+         */
+        std::array<double,2> get_surface_coordinates() const;
+
+        /**
+         * The coordinate that represents the 'depth' direction in the chosen
+         * coordinate system.
+         */
+        double get_depth_coordinate() const;
```

Comment:
value or coordinate?

Reply:
it is the depth value of the coordinate, which is to make it more complicated defined from the bottom/center outwards. Rename?